pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -195,8 +207,35 @@ func (p *adProvider) getGroupPrincipalsFromSearch(searchBase string, filter stri
 			// If bind fails because service account password has changed, just return identities formed from groups in `memberOf`
 			groupList := []v3.Principal{}
 			for _, dn := range groupDN {
+
+				search2 := ldapv2.NewSearchRequest(searchBase,
```

Comment:
This is going to make an ldap query for every group the user is a member of. The performance of this when a member is part of a large number of groups is going to be unacceptably bad.

Reply:
Btw, we are handling an error situation here  - when bind with the service account credentials fails, we just return the group principals by just setting the DN since we cannot query for other details. So in this situation, I dont think we should be querying for the uniqueId of the group either.